lib/iris/fileformats/rules.py

```diff
@@ -906,7 +906,7 @@ def _make_cube(field, converter):
                           cell_methods=metadata.cell_methods,
                           dim_coords_and_dims=metadata.dim_coords_and_dims,
                           aux_coords_and_dims=metadata.aux_coords_and_dims,
-                          fill_value=field.bmdi, dtype=field.core_data().dtype)
+                          fill_value=field.bmdi, dtype=field.data_dtype)
```

Comment:
Hm, I changed a test for this in #2557 -- we will need to look out for follow-on failures once the dust has settled between these two PRs.